### PR TITLE
libevent: update to 2.1.10

### PR DESCRIPTION
**Pull request by jerryyhom**
+ Add 'conflicts' with libev

see https://trac.macports.org/ticket/58733

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.9
Xcode 5.1

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?

### CODE REVIEW

File: devel/libevent/Portfile [hunk 1]
```diff
@@ -20,12 +21,14 @@ long_description    The libevent API provides a mechanism to execute \
 
 homepage            http://libevent.org
 
-checksums           rmd160  f23ad93d555089535085c99f3f8cbb9ebae4fd53 \
-                    sha256  d53e88f2b31fbccab93f6b1e54fe7f4c1d536b6d8250c95cd42dbc14f958baeb
+conflicts           libev
+
+checksums           rmd160 a8a158035d7677e9db00f63fab45fbadd3da41df \
+                    sha256 e864af41a336bb11dab1a23f32993afe963c1f69618bd9292b89ecf6904845b0
```
Comment (mf2k) [inline devel/libevent/Portfile]: We are now adding the `size` parameter to `checksums`.

File: devel/libevent/files/patch-apple-clang.diff [hunk 2]
```diff
@@ -0,0 +1,32 @@
+--- util-internal.h.orig	2019-02-03 13:30:49.000000000 -0800
++++ util-internal.h	2019-07-23 22:36:20.000000000 -0700
+@@ -52,16 +52,22 @@
+ 
+ /* __has_attribute() wrapper */
+ #ifdef __has_attribute
+-#define EVUTIL_HAS_ATTRIBUTE __has_attribute
++# define EVUTIL_HAS_ATTRIBUTE __has_attribute
```
Comment (pmetzger) [inline devel/libevent/files/patch-apple-clang.diff]: Why the additional space? This seems unnecessary.

File: devel/libevent/files/patch-apple-clang.diff [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
++#  endif
++# else /* !__apple_build_version__ */
++#  if __clang_major__ == 3 && __clang_minor__ >= 2 && __clang_minor__ <= 5
++#   undef EVUTIL_HAS_ATTRIBUTE
++#  endif
++# endif /* __apple_build_version__ */
++#endif /*\ defined(__clang__) && __clang__ == 1 */
+ #ifndef EVUTIL_HAS_ATTRIBUTE
+-#define EVUTIL_HAS_ATTRIBUTE(x) 0
++# define EVUTIL_HAS_ATTRIBUTE(x) 0
```
Comment (pmetzger) [inline devel/libevent/files/patch-apple-clang.diff]: Ditto here. Why the additional space?

File: devel/libevent/Portfile (hunk 1 above)
Comment (jerryyhom) [reply devel/libevent/Portfile]: Just for my curiosity, when did we start using `size`?

File: devel/libevent/files/patch-apple-clang.diff (hunk 2 above)
Comment (jerryyhom) [reply devel/libevent/files/patch-apple-clang.diff]: You are obviously correct.  In this case, the patch comes from upstream.  Would you like me to ask upstream?

File: devel/libevent/Portfile (hunk 1 above)
Comment (mf2k) [reply devel/libevent/Portfile]: It's been about a year or so.

File: devel/libevent/files/patch-apple-clang.diff (hunk 2 above)
Comment (cjones051073) [reply devel/libevent/files/patch-apple-clang.diff]: I would not bother asking upstream. The space is perfectly valid, and actually in conditional statements can make the code easier to read, for the same reasons as c,c++ is indented. In fact, common code formatting tools like clang format often do this automatically. If this is upstreams patch, we should just go with it.

File: devel/libevent/Portfile (hunk 1 above)
Comment (pmetzger) [reply devel/libevent/Portfile]: I think almost two, FWIW.

File: devel/libevent/files/patch-apple-clang.diff (hunk 2 above)
Comment (pmetzger) [reply devel/libevent/files/patch-apple-clang.diff]: Yah, it's weird that they did it but we should just use what upstream uses. I had incorrectly assumed it had originated with you.

**Pull request merged**